erpnext/subcontracting/doctype/subcontracting_order/subcontracting_order.py

```diff
@@ -169,8 +169,8 @@ def update_status(self, status=None, update_modified=True):
 				else:
 					total_required_qty = total_supplied_qty = 0
 					for item in self.supplied_items:
-						total_required_qty += item.required_qty
-						total_supplied_qty += flt(item.supplied_qty)
+						total_required_qty += item.required_qty if not item.sourced_by_supplier else 0
+						total_supplied_qty += flt(item.supplied_qty) if not item.sourced_by_supplier else 0
```

Comment:
I think this is more readable under an `if` block

```suggestion
						if not item.sourced_by_supplier:
							total_required_qty += item.required_qty
							total_supplied_qty += flt(item.supplied_qty)
```